core/services/bulletprooftxmanager/eth_broadcaster.go

```diff
@@ -207,6 +211,18 @@ func (eb *ethBroadcaster) processUnstartedEthTxs(fromAddress gethCommon.Address)
 		return errors.Wrap(err, "processUnstartedEthTxs failed")
 	}
 	for {
+		maxInFlightTransactions := eb.config.EthMaxInFlightTransactions()
+		if maxInFlightTransactions > 0 {
+			nUnconfirmed, err := CountUnconfirmedTransactions(eb.store.DB, fromAddress)
+			if err != nil {
+				return errors.Wrap(err, "CountUnconfirmedTransactions failed")
+			}
+			if nUnconfirmed >= maxInFlightTransactions {
+				logger.Warnf(`EthBroadcaster: transaction throttling; maximum number of in-flight transactions is %d per key. If this happens a lot, you might need to increase ETH_MAX_IN_FLIGHT_TRANSACTIONS. %s`, maxInFlightTransactions, utils.EthMaxInFlightTransactionsWarningLabel)
+				time.Sleep(InFlightTransactionRecheckInterval)
+				continue
```

Comment:
hmm it doesn't feel safe to continue indefinitely here - processUnstartedEthTxs is within `WithAdvisoryLock` which is maybe fine because it's per key, but also `monitorEthTxs` waits for all goroutines processing keys in parallel. So sending transactions from other keys might get blocked

Reply:
Excellent point. Might be time to have independent loops for each key